plugin/src/main/java/org/opensearch/ml/action/update_cache/UpdateModelCacheTransportAction.java

```diff
@@ -0,0 +1,125 @@
+    private final MLModelManager mlModelManager;
+    private final ClusterService clusterService;
+    private final Client client;
+    private DiscoveryNodeHelper nodeFilter;
+    private final MLStats mlStats;
+    private NamedXContentRegistry xContentRegistry;
+
+    private ModelAccessControlHelper modelAccessControlHelper;
+
+    @Inject
+    public UpdateModelCacheTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        MLModelManager mlModelManager,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        Client client,
+        DiscoveryNodeHelper nodeFilter,
+        MLStats mlStats,
+        NamedXContentRegistry xContentRegistry,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        super(
+            MLUpdateModelCacheAction.NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            MLUpdateModelCacheNodesRequest::new,
+            MLUpdateModelCacheNodeRequest::new,
+            ThreadPool.Names.MANAGEMENT,
+            MLUpdateModelCacheNodeResponse.class
+        );
+        this.mlModelManager = mlModelManager;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.nodeFilter = nodeFilter;
+        this.mlStats = mlStats;
+        this.xContentRegistry = xContentRegistry;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    @Override
+    protected MLUpdateModelCacheNodesResponse newResponse(
+        MLUpdateModelCacheNodesRequest nodesRequest,
+        List<MLUpdateModelCacheNodeResponse> responses,
+        List<FailedNodeException> failures
+    ) {
+        return new MLUpdateModelCacheNodesResponse(clusterService.getClusterName(), responses, failures);
+    }
+
+    @Override
+    protected MLUpdateModelCacheNodeRequest newNodeRequest(MLUpdateModelCacheNodesRequest request) {
+        return new MLUpdateModelCacheNodeRequest(request);
+    }
+
+    @Override
+    protected MLUpdateModelCacheNodeResponse newNodeResponse(StreamInput in) throws IOException {
+        return new MLUpdateModelCacheNodeResponse(in);
+    }
+
+    @Override
+    protected MLUpdateModelCacheNodeResponse nodeOperation(MLUpdateModelCacheNodeRequest request) {
+        return createUpdateModelCacheNodeResponse(request.getUpdateModelCacheNodesRequest());
+    }
+
+    private MLUpdateModelCacheNodeResponse createUpdateModelCacheNodeResponse(
+        MLUpdateModelCacheNodesRequest mlUpdateModelCacheNodesRequest
+    ) {
+        String modelId = mlUpdateModelCacheNodesRequest.getModelId();
+        boolean isPredictorUpdate = mlUpdateModelCacheNodesRequest.isPredictorUpdate();
+
+        Map<String, String> modelUpdateStatus = new HashMap<>();
+        modelUpdateStatus.put(modelId, "received");
```

Comment:
what's the significance of this message `received`? I didn't see anywhere to use this?

Reply:
This just follows other node level transport action pattern